### ISSUE TITLE
HACS not updating past past 1.15.2

### ISSUE THREAD

**Issue by JackPala** [hacs/integration#2553]
## System Health

version | core-2022.3.3
-- | --
installation_type | Home Assistant Container
dev | false
hassio | false
docker | true
user | root
virtualenv | false
python_version | 3.9.9
os_name | Linux
os_version | 5.4.0-96-generic
arch | x86_64
timezone | America/Bogota

<details><summary>Home Assistant Community Store</summary>

GitHub API | ok
-- | --
Github API Calls Remaining | 4903
Installed Version | 1.15.2
Stage | running
Available Repositories | 1008
Installed Repositories | 6

</details>

<details><summary>Home Assistant Cloud</summary>

logged_in | false
-- | --
can_reach_cert_server | ok
can_reach_cloud_auth | ok
can_reach_cloud | ok

</details>

<details><summary>Lovelace</summary>

dashboards | 1
-- | --
resources | 1
views | 5
mode | storage

</details>

<details><summary>Spotify</summary>

api_endpoint_reachable | ok
-- | --

</details>

<details><summary>Xiaomi Miot Auto</summary>

component_version | 0.5.3
-- | --
can_reach_server | ok
can_reach_spec | ok
logged_accounts | 1
total_devices | 1

</details>

### Checklist

- [X] I'm running the newest version of HACS <https://github.com/hacs/integration/releases/latest>
- [X] I have enabled debug logging for my installation.
- [X] I have filled out the issue template to the best of my ability.
- [X] I have read <https://hacs.xyz/docs/issues>
- [X] This issue is related to the backend (integration part) of HACS.
- [X] This issue only contains 1 issue (if you have multiple issues, open one issue for each issue).
- [x] This issue is not a duplicate issue of currently [open](https://github.com/hacs/integration/issues) or issues [pending release](https://github.com/hacs/integration/issues?q=is%3Aissue+is%3Aclosed+sort%3Aupdated-desc+milestone%3Anext).

### Describe the issue

Using the GUI to update HACS (Or any of it's frontend or backend plugins) Results in what appears in the GUI to be a successful update. The system asks for a restart of the instance as normal, however upon a reboot it appears that the previous version of HACS remains as the installed version

### Reproduction steps

1. Attempt to update
2. Restart HASS
3. System remains on old version
...

### Debug logs

```text
No logs related to Hacs appear
```

**Comment by cremor** [comment]
I have a similar issue. The first version of HACS I ever installed was 1.21.0. And that is the version I'm still on. There are no update notifications. And it seems like I also can't manually trigger an update.

I found a few things that might be related:
- The script from https://get.hacs.xyz downloads https://github.com/hacs/integration/releases/latest/download/hacs.zip which seems to only contain version 1.19.3 (I have no idea how I got from 1.19.3 to 1.21.0).
- https://github.com/hacs/integration/releases also lists 1.19.3 as the latest version.
- Only on https://github.com/hacs/integration/tags you can see that there are tags for newer versions.

**Comment by JackPala** [comment]
@cremor do you know if manually removing the HACS instance and reinstalling it will break the relationship between the HACS instance  & the child frontend/backend integrations?

**Comment by cremor** [comment]
No, I don't know that, sorry.

**Comment by cfrudolphy** [comment]
I am having same or similar issue.  Got notice the the LG ThinQ integration needed to update.  Clicked update button had spinning arrow, then notice a top of page went pink stating needed a restart.  Clicked navigate button to right of message and then clicked restart.  After restart still says I need to update LG ThinQ.

Also just installed a NetGate firewall appliance running their pfSense firewall to my network.  Went to HACS found the pfSense for Home Assistant integration.  Downloaded the integration.  Went to Configuration --> Devices & Services --> + Integration and searched for pfSense not in the list to install.  Went to file editor --> Custom_Components folder.  There is a folder for pfSense and it is populated with 8 files and one more folder named "Translations".  So HACS downloaded the integration but for what ever reason it is not registering itself with "Integrations" so it can be installed.

New behavior for me but it has been a few days since I updated anything with HACS.

Hope this is related and helps you out.

Thanks for the good work on HACS!

**Comment by teras** [comment]
I have a similar issue with @cfrudolphy 

Actually the integrations that asks me to upgrade are

*  Bodymiscale
* Passive BLE monitor integration'
* Xiaomi Gateway 3
* Tapo: Cameras Control

For me the problem is even worse, since the gateway upgrade is mandatory, since the current version broke the compatibility with latest HA and the hub stopped working :open_mouth:

**Comment by cfrudolphy** [comment]
Updated core to 2022.03.4 then tried to update LG ThinQ and install adsense for HA.  Same problems as before.  No change in behavior.

**Comment by Zugdar** [comment]
I did run the normal setup script in the SSH and that seems to have taken care of it

wget -O - https://get.hacs.xyz | bash -

**Comment by ludeeus** [comment]
HA 2022.3.X removed the loop argument compatibility layer for async_timeout.
There are 2 ways to fix that issue.
1) Run the download script again as @Zugdar mentioned.
2) Downgrade HA, upgrade HACS, then upgrade HA again.

**Comment by teras** [comment]
I'd suggest, if possible, to leave this issue open for a few days, to help people coming to this page for help, to be easy to find this solution :smile:

**Comment by cremor** [comment]
Also, please clearify if this workaround is needed only once. So was HACS already updated to handle the change in HA? Or do we need to "reinstall" for all updates for now?

**Comment by ludeeus** [comment]
HACS was adjusted a long time ago, this is one-time thing if you are on earlier version.

**Comment by cremor** [comment]
@ludeeus I just executed the download command again and that caused a HACS **downgrade** to 1.19.3.
Please see my previous comment in this issue where I describe that the download package contains version 1.19.3. I assume this is the source of the problem.

**Comment by ludeeus** [comment]
no, this issue is about 1.15.2

**Comment by cremor** [comment]
So should I open a separate issue?

**Comment by ludeeus** [comment]
if you have similar on 1.19.3 then yes.

**Comment by cfrudolphy** [comment]
I can confirm that running Zugdar's command:
`wget` -O - https://get.hacs.xyz/ | bash -`
Worked for me.  Problem solved.

**Comment by JGNi** [comment]
> I can confirm that running Zugdar's command:
> 
> `wget` -O - https://get.hacs.xyz/ | bash -`
> 
> Worked for me.  Problem solved.

You had 1.15?

**Comment by cfrudolphy** [comment]
No I was at 1.23 but after running the command I am at 1.19.3.  All I know is it works now when it didn't on 1.23
